macros/src/types/enum.rs

```diff
@@ -120,13 +134,28 @@ fn format_variant(
             },
             None => match &variant.fields {
                 Fields::Unnamed(unnamed) if unnamed.unnamed.len() == 1 => {
-                    let FieldAttr { type_override, .. } =
-                        FieldAttr::from_attrs(&unnamed.unnamed[0].attrs)?;
+                    let FieldAttr {
+                        type_as,
+                        type_override,
+                        ..
+                    } = FieldAttr::from_attrs(&unnamed.unnamed[0].attrs)?;
+
+                    if let (Some(_type_as), Some(_type_override)) = (&type_as, &type_override) {
+                        syn_err!("`type` is not compatible with `as`")
+                    }
+
+                    let parsed_ty = if let Some(_type_as) = &type_as {
```

Comment:
`_type_as` in not unused, so it should not have an underscore. If you are worried about name collisions and shadowing, give the variable a diferent name

Also, to prevent the move, instead of
```rs
if let Some(type_as) = &type_as {
```

you can probably use
```rs
if let Some(ref type_as) = type_as {
```

I haven't checked this against the compiler, but if it does compile I think you should write it this way